alpha/alpha-fsm-channel-kafka/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/kafka/KafkaMessagePublisher.java

```diff
@@ -43,10 +44,15 @@ public void publish(Object data) {
         }
 
         try {
-            kafkaTemplate.send(topic, data).get();
-        } catch (InterruptedException | ExecutionException e) {
+            if(data instanceof BaseEvent) {
+                BaseEvent event = (BaseEvent) data;
+                kafkaTemplate.send(topic, event.getGlobalTxId(), event).get();
```

Comment:
Why do we need to call the Future get method?
 I don't think we need to make the async invocation to be a sync invocation.

Reply:
> Why do we need to call the Future get method?
> I don't think we need to make the async invocation to be a sync invocation.

We need to make sure the message is delivered, When the transmission fails, we need to throw an exception to the omega side.

Reply:
It will introduce some performance issue,  we should revisit it later.

Reply:
> It will introduce some performance issue, we should revisit it later.

I think need a trade-off in reliability and performance, I sent a message to discuss this issue 
https://lists.apache.org/thread.html/f4bb944c2d6b3ccd55fe8c0ad1e1d000e94bba09277ed842a1dff480@<dev.servicecomb.apache.org>

https://lists.apache.org/thread.html/f4bb944c2d6b3ccd55fe8c0ad1e1d000e94bba09277ed842a1dff480@%3Cdev.servicecomb.apache.org%3E